app/controllers/concerns/authentication.rb

```diff
@@ -102,5 +109,9 @@ def create_or_fetch_authed_user(auth_data)
           end
       end
     end
+
+    def root_path_with_params
+      root_path(last_course_id: session[:last_course_id], just_signed_in: !session[:user].nil? ? true : nil)
```

Comment:
```suggestion
      root_path(last_course_id: session[:last_course_id], just_signed_in: session[:user].present?)
```

Reply:
Need it to return nil rather than false so that the just_signed_in param isn't added.

Reply:
Ah yes, missed that.